### PR TITLE
added a few Dart and Flutter online courses

### PR DESCRIPTION
**Pull request by its-me-mahmud**
## What does this PR do?
Add resource(s)

### Description
Add a few Dart and Flutter online courses from YouTube

### For book lists, is it a book? For course lists, is it a course? etc.
It's a course.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -303,6 +303,7 @@
 * [Dart Programming in 4 hours \| Full beginners tutorial](https://www.youtube.com/watch?v=5xlVP04905w) - Mike Dane (YouTube)
 * [Dart Programming Tutorial - Full Course](https://www.youtube.com/watch?v=Ej_Pcr4uC2Q) - Mahmud Ahsan (FreeCodeCamp)
 * [Dart Programming Tutorial \| Learn the Dart for Flutter](https://www.udemy.com/course/dart-programming-tutorial-learn-the-dart-for-flutter) (Udemy)
+* [Free Dart Course 💙](https://www.youtube.com/playlist?list=PL6yRaaP0WPkVR2FiAS7TTCT_n2mDhwISE) - Vandad Nahavandipoor (YouTube)
```
Comment (LuigiImVector) [inline courses/free-courses-en.md:306]:
```suggestion
* [Free Dart Course](https://www.youtube.com/playlist?list=PL6yRaaP0WPkVR2FiAS7TTCT_n2mDhwISE) - Vandad Nahavandipoor (YouTube)
```

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -390,8 +391,10 @@
 ### Flutter
 
 * [Flutter Crash Course](https://fluttercrashcourse.com/courses/basics/lessons/materialapp-scaffold-appbar-text) - Nick Manning *(account required)*
+* [Flutter State Management Course 💙](https://www.youtube.com/playlist?list=PL6yRaaP0WPkUf-ff1OX99DVSL1cynLHxO) - Vandad Nahavandipoor (YouTube)
 * [Flutter Tutorial](https://www.udacity.com/course/build-native-mobile-apps-with-flutter--ud905) - Matt Sullivan, James Williams, Mary Xia (Udacity)
 * [Flutter Tutorial for Beginners](https://www.udemy.com/course/learn-flutter-beginners-course) - Mayuresh Wankhede (Udemy)
+* [Free Flutter Course 💙](https://www.youtube.com/playlist?list=PL6yRaaP0WPkVtoeNIGqILtRAgd3h2CNpT) - Vandad Nahavandipoor (YouTube)
```
Comment (LuigiImVector) [inline courses/free-courses-en.md:397]:
```suggestion
* [Flutter State Management Course](https://www.youtube.com/playlist?list=PL6yRaaP0WPkUf-ff1OX99DVSL1cynLHxO) - Vandad Nahavandipoor (YouTube)
* [Flutter Tutorial](https://www.udacity.com/course/build-native-mobile-apps-with-flutter--ud905) - Matt Sullivan, James Williams, Mary Xia (Udacity)
* [Flutter Tutorial for Beginners](https://www.udemy.com/course/learn-flutter-beginners-course) - Mayuresh Wankhede (Udemy)
* [Free Flutter Course](https://www.youtube.com/playlist?list=PL6yRaaP0WPkVtoeNIGqILtRAgd3h2CNpT) - Vandad Nahavandipoor (YouTube)
```

**Review by LuigiImVector — COMMENTED** [review]
it is recommended to avoid emoji for uniform reading

**Comment by its-me-mahmud** [comment]
> it is recommended to avoid emoji for uniform reading

There wasn't mention of this anywhere in the contributing guidelines. So hope you'll include that in the docs so that others can know about this.

**Comment by its-me-mahmud** [comment]
https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md#in-a-nutshell
Here on the number 5 
>GitHub Actions will run tests to make sure your lists are alphabetized and formatting rules are followed. Be sure to check that your changes pass the tests.

It wasn't clear and that's why I made a mistake. Also, I saw others make the same mistake.

#### How It's alphabetized?

#### Is it only with the first letter of a word?

#### Or is it with each and every word?

So hope you'll include more specific guides on this.

**Comment by LuigiImVector** [comment]
I will edit the CONTRIBUTING file by adding these things.

**Pull request closed without merging**